[FredericChow00] iP

FredericChow00 opened the pull request:
# DukeBaby

> Due tomorrow do tomorrow - CHS 4-1 2016

DukeBaby is the personal assistant that everybody wants and needs to stop them from procrastinating

* Easy to learn
* Easy to use
* ~~Easy~~ _VERY EASY_ way to cure procrastination

All you need to do is:
1. Download from [here](https://github.com/FredericChow00/ip/releases).
1. Double-click it.
1. Add your tasks
1. ~~Start~~ Stop your procrastinating journey! :tada:

Best of all, it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon!)
- [ ] Reminders (coming soon!)

If you are a Java programmer, here is the `main` method:

``` ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

FredericChow00 committed:
Level-1 Completed

FredericChow00 committed:
Level-2 Completed

FredericChow00 committed:
Edited Level-2

FredericChow00 committed:
Level-3 and A-classes Completed

FredericChow00 committed:
Level-4 and A-Inheritance Completed

FredericChow00 committed:
A-TextUiTesting Completed

FredericChow00 committed:
Level-5 and A-Exceptions Completed

FredericChow00 committed:
Level-6 and A-Collections Completed

FredericChow00 committed:
A-Enums Completed

FredericChow00 committed:
Updated minor warnings

FredericChow00 committed:
Updated non-numeric exception handlings

FredericChow00 committed:
Level-7 Completed

FredericChow00 committed:
Edited Level-7

FredericChow00 committed:
Level-8 Minimal

FredericChow00 committed:
Please enter the commit message for your changes. Lines starting

FredericChow00 committed:
Merge branch 'master' into branch-level-8

FredericChow00 committed:
Update File Pathing

FredericChow00 committed:
Merge branch 'add-gradle-support'

FredericChow00 committed:
Complete A-Gradle Minimal

FredericChow00 committed:
Complete A-MoreOOP Minimal

FredericChow00 committed:
Complete A-Packages

FredericChow00 committed:
Complete A-Packages

FredericChow00 committed:
Complete A-Unit

FredericChow00 committed:
Complete A-Jar

FredericChow00 committed:
Add MANIFEST.MF

FredericChow00 committed:
Add MANIFEST.MF

FredericChow00 committed:
Complete A-JavaDoc

FredericChow00 committed:
Add more JavaDocs

FredericChow00 committed:
Complete A-CodingStandard

FredericChow00 committed:
"Merge A-JavaDoc and A-CodingStandard to master"

FredericChow00 committed:
Edit Coding Standards and JavaDocs

FredericChow00 committed:
"Edit some files"

FredericChow00 committed:
Complete Level-9

FredericChow00 committed:
"Complete Level-9"

FredericChow00 committed:
Merge Level-9

FredericChow00 committed:
Merge branch 'branch-level-8'

FredericChow00 committed:
Change build.gradle

FredericChow00 committed:
Complete A-Checkstyle

FredericChow00 committed:
Complete Level-10

FredericChow00 committed:
Edit GUI Aesthetics

WindyCall commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+
+    private static final String SAVE_FILE_NAME = "duke.txt";
+
+    enum Commands {
+        list, bye, todo, mark, unmark, event, deadline, delete, find
+    }
```
Will this be better if you use a separate Enumeration class instead of putting it inside your main Duke class?

FredericChow00 replied:
Hmm yeah that's true although I'm not sure if my Enum class will have any methods though. Thanks for the advice!

WindyCall commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,262 @@
+                String keywords = getKeywords(inputStrings);
+                TaskList matchingTasks = taskList.findTask(keywords);
+                output = ui.sayMatchingTasks(matchingTasks);
+                break;
```
Very good follow of switch/case coding standard!!

WindyCall commented on src/main/java/seedu/duke/TaskList.java:
```diff
@@ -0,0 +1,143 @@
+package seedu.duke;
+
+import seedu.duke.tasks.*;
```
Should explicitly importing all classes needed better as required by given Java coding standard?

FredericChow00 replied:
Yup will change this! I used * because I thought that if there's more than 3 imports from the same package we can just replace with * but i guess not hahaha

WindyCall reviewed (commented):
Well done!! I think your variable namings are generally very well considered and self-explanatory. What impressed me most is how your format corresponds to the requirement and your good overall OOP construction. Nice to see so excellent work!!!

jytheawesome commented on src/main/java/seedu/duke/Main.java:
```diff
@@ -0,0 +1,33 @@
+package seedu.duke;
+
+import java.io.IOException;
+
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Scene;
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+import seedu.duke.gui.MainWindow;
+
```
Do you need to separate imports of different packages by a blank line here? Correct me if I'm wrong

jytheawesome commented on src/main/java/seedu/duke/Duke.java:
```diff
@@ -0,0 +1,47 @@
+package seedu.duke;
+
+public class Duke {
+    private final Storage storage;
+    private final Ui ui;
+    private final Parser parser;
+    private TaskList tasks;
```
good use of access modifiers!

jytheawesome reviewed (commented):
I liked your naming conventions! They were clear and easily understandable. Good job!

FredericChow00 committed:
Modify build.gradle for creation of new JAR file

Previous JAR file was outdated

Changed the mainClassName so that the newly incorporated GUI using JavaFX can be included in the JAR file.

Allows users to:
* Refer to GUI for easier usage of Duke.
* Type on text box in GUI instead of CLI commands.

FredericChow00 committed:
Complete A-Assertions

There are some methods that did not have assertions to ensure that the code works as intended.

Add assertions to several methods to ensure that the code logic is working as intended.

Allows developers to:
- Have more layers of checks to ensure that code is working as intended.
- Debug any issues easily

FredericChow00 committed:
Merge pull request #1 from FredericChow00/branch-A-Assertions

Complete A-Assertions

FredericChow00 committed:
Complete A-CodeQuality

There were several methods that spanned up to 80 lines of code that makes it difficult for developers to read and understand the code.

Abstract repetitive code and follow SLAP principles to enhance quality of code.

Allows developers to:
* Have a better time trying to understand the code
* Debug any issues easily

FredericChow00 committed:
Merge branch 'master' into branch-A-CodeQuality

FredericChow00 committed:
Merge pull request #2 from FredericChow00/branch-A-CodeQuality

Complete A-CodeQuality

FredericChow00 committed:
Remove magic numbers

FredericChow00 committed:
Remove use of wildcard in import

FredericChow00 committed:
Complete A-CI

FredericChow00 committed:
B-ViewSchedules

FredericChow00 committed:
Add Ui Screenshot

FredericChow00 committed:
Complete A-UserGuide

FredericChow00 committed:
Merge pull request #4 from FredericChow00/branch-A-UserGuide

Complete A-UserGuide

FredericChow00 committed:
Complete A-UserGuide

FredericChow00 committed:
Merge branch 'master' of https://github.com/FredericChow00/ip

FredericChow00 committed:
Fix Java Version Warning

FredericChow00 committed:
Edit Storage pathing

FredericChow00 committed:
Edit Storage

FredericChow00 committed:
Edit README.md
